plugins/discourse-local-dates/assets/javascripts/initializers/discourse-local-dates.js

```diff
@@ -128,6 +151,53 @@ function initializeDiscourseLocalDates(api) {
       },
     },
   });
+
+  let startRangeOpts = null;
```

Comment:
This feels like a "global" being used by two callback functions. I wonder if we can pass this between the callback functions instead with some kind of metadata object.

Reply:
Yeah I don't like it a whole lot, maybe I can see if I can pass through some sort of metadata object in all of these `to-markdown` callbacks? Something that is kept in memory through the whole `toMarkdown()` chain for one block of HTML defined here https://github.com/discourse/discourse/blob/7120a33398998a07359330bc1fd703096585ed87/app/assets/javascripts/discourse/app/lib/to-markdown.js#L866 ? 

So then you would get something like

```
const metadata = {};
for (const callback of blockDecorateCallbacks) {
  const result = callback.call(this, metadata, text);

  if (typeof result !== "undefined") {
    text = result;
  }
}
```

I am not sure if I will be able to make this data flow through the whole way though....maybe I can attach arbitrary metadata to the `next/previous` Element instead?

Reply:
Was able to do this by passing through a `metadata` object through the flows